### DISCUSSION TITLE
Just a matter of curiosity

### DISCUSSION THREAD

**Discussion by pas-comme** [nordic-institute/X-Road#2223] [Q&A]
Dear all,

I would like to use X-Road to exchange data between multiple information systems.

I have a question out of curiosity regarding security: if a malicious actor gains full control of an information system within the X-Road ecosystem, explain the principles of the ecosystem that prevent this from affecting other information systems.

Thank you to anyone who responds

Best regards,

**Comment by raits** [comment]
Hello @pas-comme,

There are several different options depending on which level the problem needs to be handled. For information systems you have at least the following options:

- On a local firewall level you could just not allow the compromised system to connect to the Security Server
- You could disable the subsystem from the Security Server UI
- You could change the client certificate the subsystem uses to communicate, or if they don't use a certificate you could switch the connection to require one
- You could unregister the subsystem from the Security Server from the Security Server UI
- The X-Road operator could unregister the subsystem from the Central Server UI
- The other subsystems on the instance that have allowed access to services could remove access rights for that system
- The X-Road operator could remove the Security Server or member from the instance
- The CA could invalidate the signing certificate or even the authentication certificate

These are just some of the options on various levels, but it really depends on how far it is compromised.
